languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -259,60 +560,266 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
+}
+
+(switch_block (switch_block_statement_group (switch_label)+ . (statement) @first)) @this {
+  edge @first.before_scope -> @this.lexical_scope
+}
+
+(switch_block (switch_block_statement_group (switch_label)+ (statement) @a . (statement) @b)) @this {
+  edge @b.before_scope -> @a.after_scope
+}
+
+(switch_block (switch_rule (switch_label) @label (_) @body)) @this {
+  edge @label.lexical_scope -> @this.lexical_scope
+  edge @body.before_scope -> @this.lexical_scope
+}
+
+(switch_label) @label {
+  node @label.lexical_scope
+}
+
+(switch_label (expression) @expr) @label {
+  edge @expr.lexical_scope -> @label.lexical_scope
+}
+
+(synchronized_statement (_) @expr body: (_) @body) @stmt {
+  edge @expr.lexical_scope -> @stmt.before_scope
+  edge @body.before_scope -> @stmt.before_scope
+  edge @stmt.after_scope -> @stmt.before_scope
+}
+
+(try_statement body: (_) @body) @stmt {
+  edge @body.before_scope -> @stmt.before_scope
+  edge @stmt.after_scope -> @stmt.before_scope
+}
+
+(try_statement (catch_clause (catch_formal_parameter (catch_type) @type (_) @id) body: (_) @body)) @stmt {
+  node def
+  node defs
+  attr (def) node_definition = @id
+  edge defs -> def
+  edge @body.before_scope -> defs
+  edge defs -> @stmt.before_scope
+
+  edge @type.lexical_scope -> @stmt.before_scope
+
+  node def__typeof
+  attr (def__typeof) pop_symbol = ":"
+
+  edge def -> def__typeof
+
+  edge def__typeof -> @type.type
+}
+
+(catch_type) @catch_type {
+  node @catch_type.type
+  node @catch_type.lexical_scope
+}
+
+(catch_type (_) @type) @catch_type {
+  edge @catch_type.type -> @type.type
+  edge @type.lexical_scope -> @catch_type.lexical_scope
+}
+
+(try_statement (finally_clause (_) @finally)) @stmt {
+  edge @finally.before_scope -> @stmt.before_scope
+}
+
+(try_with_resources_statement) @stmt {
+  edge @stmt.after_scope -> @stmt.before_scope
+}
 
-  node @type.value
-  attr(@type.value) node_reference = @type
+(try_with_resources_statement resources: (resource_specification . (resource) @first)) @stmt {
+  edge @first.before_scope -> @stmt.before_scope
+}
+
+(try_with_resources_statement resources: (resource_specification (resource) @a . (resource) @b)) @stmt {
+  edge @b.before_scope -> @a.after_scope
+}
+
+(try_with_resources_statement resources: (resource_specification (resource) @last .) body: (_) @body) @stmt {
+  edge @body.before_scope -> @last.after_scope
+}
+
+(resource) @this {
+  node @this.before_scope
+  node @this.after_scope
+}
+
+(resource type: (_) @type (_) @id value: (_) @value) @this {
+  edge @type.lexical_scope -> @this.before_scope
+  edge @value.lexical_scope -> @this.before_scope
+
+  node def
+  attr (def) node_definition = @id
+
+  node def__typeof
+  attr (def__typeof) pop_symbol = ":"
+
+  edge def -> def__typeof
+
+  edge def__typeof -> @type.type
+
+  edge @this.after_scope -> def
+  edge @this.after_scope -> @this.before_scope
+}
+
+(resource . (identifier) @id .) @this {
```

Comment:
We should probably take a pass through and add field names to the grammar for cases like this, but I consciously wanted to avoid editing the grammar for the moment.